backend/instancemgmt/instance_manager.go

```diff
@@ -109,35 +128,28 @@ func (im *instanceManager) Get(ctx context.Context, pluginContext backend.Plugin
 		}
 	}
 
-	im.locker.Lock(cacheKey)
-	defer im.locker.Unlock(cacheKey)
+	im.locker.Lock(strKey)
+	defer im.locker.Unlock(strKey)
 
-	if item, ok := im.cache.Load(cacheKey); ok {
+	if item, ok := im.cache.Get(strKey); ok {
 		ci := item.(CachedInstance)
 		needsUpdate := im.provider.NeedsUpdate(ctx, pluginContext, ci)
 
 		if !needsUpdate {
 			return ci.instance, nil
 		}
 
-		if disposer, valid := ci.instance.(InstanceDisposer); valid {
-			time.AfterFunc(disposeTTL, func() {
-				disposer.Dispose()
-				activeInstances.Dec()
-			})
-		} else {
-			activeInstances.Dec()
-		}
+		im.cache.Delete(strKey)
```

Comment:
It may be a little faster to avoid this delete, make a function for the disposer code, and call that function from here and in the OnEvicted handler.